multiplayer/src/components/ArcadeSimulator.tsx

```diff
@@ -96,68 +100,70 @@ export default function Render() {
         return () => window.removeEventListener("message", msgHandler);
     }, [clientRole]);
 
-    const getOpts = () => {
-        const opts: pxt.runner.SimulateOptions = {
-            embedId: "multiplayer-sim",
-            additionalQueryParameters: selectedPlayerTheme,
-            single: true,
-            autofocus: true,
-            fullScreen: true,
-            /** Enabling debug mode so that we can stop at breakpoints as a 'global pause' **/
-            debug: true,
-            mute: state.muted,
-        };
+    const getOpts: () => RunOptions = () => {
+        let opts: RunOptions;
 
         if (isHost) {
-            opts.id = gameId;
-            opts.mpRole = "server";
+            opts = {
+                simQueryParams: selectedPlayerTheme,
+                mpRole: "server",
+                id: gameId!,
+            };
         } else {
-            opts.code = "multiplayer.init()";
-            opts.mpRole = "client";
+            opts = {
+                simQueryParams: selectedPlayerTheme,
+                mpRole: "client",
+            };
         }
 
         return opts;
     };
 
     const compileSimCode = async () => {
-        builtSimJsInfo = pxt.runner.buildSimJsInfo(getOpts());
+        builtSimJsInfo = buildSimJsInfo(getOpts());
         return await builtSimJsInfo;
     };
 
-    const preloadSim = async () => {
-        pxt.runner.preloadSim(simContainerRef.current!, getOpts());
-    };
-
     const runSimulator = async () => {
         const simOpts = getOpts();
         if (builtSimJsInfo) {
             simOpts.builtJsInfo = await builtSimJsInfo;
         }
 
-        builtSimJsInfo = pxt.runner.simulateAsync(
-            simContainerRef.current!,
-            simOpts
-        );
+        builtSimJsInfo = simulateAsync(simContainerRef.current!, simOpts);
 
         await builtSimJsInfo;
     };
 
     useEffect(() => {
-        if (gameState?.gameMode === "playing") {
+        if (gameMode === "playing") {
             runSimulator();
         }
-    }, [gameState]);
+    }, [gameMode]);
 
     useEffect(() => {
-        const codeReadyToCompile =
-            playerSlot! > 1 || (playerSlot == 1 && gameId);
-        if (codeReadyToCompile && gameState?.gameMode !== "playing") {
-            preloadSim().then(compileSimCode);
+        if (gameState?.gameMode !== "playing" && simContainerRef.current) {
```

Comment:
```suggestion
        if (gameMode !== "playing" && simContainerRef.current) {
```
it's the smallest suggestion I could come up with